engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/alias/graph/GraphImpl.scala

```diff
@@ -102,7 +102,7 @@ sealed private[graph] class GraphImpl(
   final def resolveLocalUsage(
     occurrence: GraphOccurrence.Use
   ): Option[Graph.Link] = {
-    scopeFor(occurrence.id).flatMap(_.resolveUsage(occurrence).map { link =>
+    Option(occurrence.scope()).flatMap(_.resolveUsage(occurrence).map { link =>
```

Comment:
Each `Use` is allocated at a particular `Scope`. By keeping a reference from `Use` to `Scope` we shall speed the  [ManyLocalVarsBenchmark.longMethodWithLotOfLocalVars](https://github.com/enso-org/enso/actions/runs/14222861579) up by 25%(?).